src/Markdig/Extensions/Emoji/EmojiMapping.cs

```diff
@@ -0,0 +1,1784 @@
+// Copyright (c) Alexandre Mutel. All rights reserved.
+// This file is licensed under the BSD-Clause 2 license. 
+// See the license.txt file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Markdig.Helpers;
+
+namespace Markdig.Extensions.Emoji
+{
+    /// <summary>
+    /// An emoji and smiley mapping, to be used by <see cref="EmojiParser"/>.
+    /// </summary>
+    public class EmojiMapping
+    {
+        /// <summary>
+        /// The default emojis and smileys mapping.
+        /// </summary>
+        public static readonly EmojiMapping DefaultEmojiMapping = new EmojiMapping();
+
+        internal CompactPrefixTree<string> EmojiPrefixTree { get; }
+        internal CompactPrefixTree<string> EmojiSmileyPrefixTree { get; }
+        internal char[] EmojiOpeningCharacters { get; }
+        internal char[] EmojiSmileyOpeningCharacters { get; }
+
+        #region Emojis and Smileys
+
+        /// <summary>
+        /// Returns a new instance of the default emoji to unicode dictionary.
+        /// It can be used to create a customized <see cref="EmojiMapping"/>.
+        /// </summary>
+        public static IDictionary<string, string> GetDefaultEmojiToUnicode()

```

Comment:
This can be Dictionary instead of the interface

Reply:
Isn't it better to use the interface when possible and where it makes sense?

I used the interface because the dictionary can still be used (elements added and modified) with `IDictionary`, and it matches the type in the ctor of `EmojiMapping`.

Reply:
Did you miss my comment? Any thought?

Reply:
Ah yeah I missed it. I suppose it doesn't really make a difference in this case.